modules/core/src/webgl/utils/request-animation-frame.js

```diff
@@ -1,14 +1,14 @@
 // Node.js polyfills for requestAnimationFrame and cancelAnimationFrame
 /* global window, setTimeout, clearTimeout */
 
-export function requestAnimationFrame(callback) {
-  return typeof window !== 'undefined' && window.requestAnimationFrame
-    ? window.requestAnimationFrame(callback)
+export function requestAnimationFrame(callback, device = window) {
```

Comment:
This is risky. Polyfills should mimic the API of the functions they're replacing. For example, if Luma is included in a codebase that has already polyfilled rAF, this custom version will never be defined.

Reply:
alternative suggestion?

Reply:
You don't need a polyfill for `device.requestAnimationFrame` so you can just use that function directly in the VR case. One option, if you need to override the internal rAF, is to just have the function to use passed in as a parameter to the `AnimationLoop` constructor. Or you could subclass `AnimationLoop` for the VR case and have the override happen there.

Reply:
Did it.